handler/handler.go

```diff
@@ -640,3 +659,51 @@ func switchToIsClassifiedBy(toChange []interface{}) []interface{} {
 	}
 	return changed
 }
+
+func isAuthorizedForDelete(r *http.Request, scheduledForDelete interface{}) bool {
```

Comment:
`comment`: Unfortunately we didn't write a middleware which calls a specific handle based on the opa evaluation, if we had such, you would have been able to get the annotations, use opa for the access control instead of this custom function, and later call the handler to actually delete the resource, we can think about this in the future.

Reply:
Maybe, but if you call the handler in a middleware the annotation will be already deleted

Reply:
No, no, maybe I didn't make myself clear, imagine a response middleware which only gets the annotation, evaluates if it is eligible for deletion and after that is able to call another handler that deletes the annotation itself.